core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala

```diff
@@ -305,10 +305,30 @@ object ReassignPartitionsCommand extends Logging {
   private def checkIfPartitionReassignmentSucceeded(zkUtils: ZkUtils, partitionsToBeReassigned: Map[TopicAndPartition, Seq[Int]])
   :Map[TopicAndPartition, ReassignmentStatus] = {
     val partitionsBeingReassigned = zkUtils.getPartitionsBeingReassigned().mapValues(_.newReplicas)
-    partitionsToBeReassigned.keys.map { topicAndPartition =>
-      (topicAndPartition, checkIfPartitionReassignmentSucceeded(zkUtils, topicAndPartition, partitionsToBeReassigned,
-        partitionsBeingReassigned))
-    }.toMap
+    val (reassigned, beingReassigned) = partitionsToBeReassigned.keys.partition { topicAndPartition =>
```

Comment:
Should the first entry in the tuple be `beingReassigned`? Also, it is a bit confusing to say `reassigned` because it suggests that the reassignment is successfully completed. It may be better to be `notBeingReassigned`.

Also, would it be simpler to optimize the existing checkIfPartitionReassignmentSucceeded(...) method instead of adding this new one?